.github/workflows/build_test.yaml

```diff
@@ -0,0 +1,57 @@
+name: Build_Test
+
+on: [push, pull_request]
+
+defaults:
+  run:
+    shell: bash
+
+env: 
+  PYTHON_VERSION: 3.8
+
+jobs:
+  smoke_test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: recursive
+        fetch-depth: 0
+    - uses: actions/setup-python@v2
+      with:
+        python-version: ${{ env.PYTHON_VERSION }}
+    - uses: ./.github/actions
+ 
+  basic:
+    needs: smoke_test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8, 3.9]
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: recursive
+        fetch-depth: 0
+    - name: Install Python
```

Comment:
There's a conda action available too

Reply:
I'm just wrapping my head around it. I'll improve it once I have a semi working thing